src/main/java/org/terasology/las/yinyang/YinYangProvider.java

```diff
@@ -5,35 +5,56 @@
 import org.joml.Vector3i;
 import org.terasology.engine.world.block.BlockRegion;
 import org.terasology.engine.world.generation.Border3D;
+import org.terasology.engine.world.generation.Facet;
 import org.terasology.engine.world.generation.FacetProviderPlugin;
 import org.terasology.engine.world.generation.GeneratingRegion;
 import org.terasology.engine.world.generation.Produces;
+import org.terasology.engine.world.generation.Updates;
+import org.terasology.engine.world.generation.facets.SurfacesFacet;
 import org.terasology.engine.world.generator.plugin.RegisterPlugin;
+import org.terasology.ligthandshadow.componentsystem.LASUtils;
 
 import java.util.Collections;
 import java.util.List;
 
 @RegisterPlugin
 @Produces(YinYangFacet.class)
+@Updates(@Facet(value = SurfacesFacet.class))
 public class YinYangProvider implements FacetProviderPlugin {
 
-    private List<Vector3i> yinYangPositions = Collections.singletonList(new Vector3i(0, 10, 0));
+    private static final int RADIUS = 5;
+    private final List<Vector3i> yinYangPositions = Collections.singletonList(new Vector3i(0, 10, 0));
 
     @Override
     public void process(GeneratingRegion region) {
         Border3D border =
                 region.getBorderForFacet(YinYangFacet.class)
                         .extendBy(0, 0, 10);
 
+        SurfacesFacet surfacesFacet = region.getRegionFacet(SurfacesFacet.class);
+
         YinYangFacet yinYangFacet = new YinYangFacet(region.getRegion(), border);
 
         BlockRegion worldRect = yinYangFacet.getWorldRegion();
 
         yinYangPositions.stream()
                 .filter(worldRect::contains)
-                .forEach(pos -> yinYangFacet.setWorld(pos, new YinYang()));
+                .forEach(pos -> {
+                    yinYangFacet.setWorld(pos, new YinYang());
+                    int y = surfacesFacet.getNextBelow(pos);
+                    for (int i = -RADIUS; i <= RADIUS; i++) {
+                        for (int j = -2 * RADIUS; j <= 2 * RADIUS; j++) {
+                            if (!LASUtils.pixel(j, i, RADIUS).equals("engine:air")) {
+                                Vector3i blockPos = new Vector3i(i, 0, j).add(pos);
```

Comment:
The yin yang symbol is not that big, so the impact of creating new vectors here might not be that big, but it's still good practice to create a new instance once outside the loop (e.g., before `yinYangPositions.stream()`) and then `set` this temporary vector here.

We could also directly write the result of the addition to that temporary vector, like so:
```suggestion
                                pos.add(i, 0, j, tempPos);
```

Reply:
https://www.baeldung.com/java-8-lambda-expressions-tips#mutation